nemo/collections/multimodal/models/multimodal_llm/neva/neva_model.py

```diff
@@ -235,6 +240,22 @@ def replace_media_embeddings(self, input_ids, inputs_embeds, media):
 
         return updated_input_embeds
 
+    def sharded_state_dict(self, prefix: str = '', sharded_offsets: tuple = (), **kwargs):
+        sharded_state_dict = super().sharded_state_dict(prefix=prefix, sharded_offsets=sharded_offsets, **kwargs)
+
+        state_dict = self.state_dict(prefix='', keep_vars=True)
+        state_dict.pop('weight')
+        # duplicate everything else
+        for layer_name in state_dict.keys():
+            tensor = state_dict[layer_name]
+            layer_key = f'{prefix}{layer_name}'
+            sharded_state_dict[layer_key] = make_sharded_tensor_for_checkpoint(
+                tensor,
+                layer_key,
+                prepend_offsets=sharded_offsets,
+            )
```

Comment:
I think this can be replaced with 
```
from megatron.core.transformer.utils import make_sharded_tensors_for_checkpoint
     ...
     make_sharded_tensors_for_checkpoint(state_dict)
```